### PR TITLE
Add optional mask to KL loss

### PR DESCRIPTION
**Pull request by lukaribar**
Small PR to add an additional output masking to the `_compute_kl_loss` function; by default it will use `attention_mask` as before if nothing is passed.

### CODE REVIEW

**Review by DouglasOrr — APPROVED** [review]
LGTM, thanks for this!
